test/extended/prometheus/prometheus.go

```diff
@@ -36,6 +36,120 @@ import (
 	helper "github.com/openshift/origin/test/extended/util/prometheus"
 )
 
+var _ = g.Describe("[sig-instrumentation][Late] Alerting rules", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLIWithoutNamespace("prometheus")
+
+		url, bearerToken string
+	)
+	g.BeforeEach(func() {
+		var ok bool
+		url, _, bearerToken, ok = helper.LocatePrometheus(oc)
+		if !ok {
+			e2e.Failf("Prometheus could not be located on this cluster, failing prometheus test")
+		}
+	})
+
+	g.It("should have a severity label equal to critical, warning or info except for Watchdog", func() {
+		if len(os.Getenv("TEST_UNSUPPORTED_ALLOW_VERSION_SKEW")) > 0 {
+			e2eskipper.Skipf("Test is disabled to allow cluster components to have different versions, and skewed versions trigger multiple other alerts")
+		}
+		ns := oc.SetupNamespace()
+		execPod := exutil.CreateExecPodOrFail(oc.AdminKubeClient(), ns, "execpod")
+		defer func() {
+			oc.AdminKubeClient().CoreV1().Pods(ns).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
+		}()
+
+		url = fmt.Sprintf("%s/api/v1/rules", url)
+		contents, err := helper.GetBearerTokenURLViaPod(ns, execPod.Name, url, bearerToken)
+		if err != nil {
+			e2e.Failf("unable to query %s: %v", url, err)
+		}
+
+		var result struct {
+			Status string        `json:"status"`
+			Data   ruleDiscovery `json:"data"`
+		}
+		if err := json.Unmarshal([]byte(contents), &result); err != nil {
+			e2e.Failf("unable to parse response %q from %s: %v", contents, url, err)
+		}
+
+		severityRe := regexp.MustCompile("^critical|warning|info$")
+
+		for _, rg := range result.Data.RuleGroups {
+			for _, r := range rg.Rules {
+				ar, ok := r.(*alertingRule)
+				if !ok {
+					continue
+				}
+
+				if ar.Name == "Watchdog" {
+					continue
+				}
+
+				severity, found := ar.Labels["severity"]
+				if !found {
+					e2e.Failf("alert %q should have a 'severity' label", ar.Name)
```

Comment:
This fails right away, vs showing all problems.  I might suggest adapting the structure in https://github.com/openshift/origin/pull/26020/files#diff-cb149e746da7ca315f8c9b852dd6f407e04e16a7199587be3ae745fa57fe681eR67 which we are using in more tests now to accumulate failures and report them.  that also gives you the ability to define the exceptions for now and merge (without having to chase the miscreants)

Reply:
yep I've realized this later on :)